enrollment-server/src/test/java/com/wultra/app/enrollmentserver/impl/service/converter/MobileTokenConverterTest.java

```diff
@@ -0,0 +1,160 @@
+        assertEquals("Můžete se stát obětí útoku.", preApprovalScreen.getMessage());
+        assertEquals(PreApprovalScreen.ApprovalType.SLIDER, preApprovalScreen.getApprovalType());
+        assertNotNull(preApprovalScreen.getItems());
+
+        final List<String> items = preApprovalScreen.getItems();
+        assertEquals(3, items.size());
+        assertEquals("Aktivujete novou aplikaci a povolujete přístup k Vašim účtům", items.get(0));
+    }
+
+    @Test
+    void testConvertUi_riskFlags_X() throws Exception {
+        final OperationDetailResponse operationDetail = new OperationDetailResponse();
+        operationDetail.setSignatureType(List.of(SignatureType.KNOWLEDGE));
+        operationDetail.setStatus(OperationStatus.APPROVED);
+        operationDetail.setRiskFlags("X");
+
+        final OperationTemplateEntity operationTemplate = new OperationTemplateEntity();
+
+        final Operation result = tested.convert(operationDetail, operationTemplate);
+
+        assertNotNull(result.getUi());
+
+        final UiExtensions ui = result.getUi();
+        assertEquals(true, ui.getFlipButtons());
+        assertNull(ui.getBlockApprovalOnCall());
+        assertNull(ui.getPreApprovalScreen());
+    }
+
+    @Test
+    void testConvertUi_riskFlags_C() throws Exception {
+        final OperationDetailResponse operationDetail = new OperationDetailResponse();
+        operationDetail.setSignatureType(List.of(SignatureType.KNOWLEDGE));
+        operationDetail.setStatus(OperationStatus.APPROVED);
+        operationDetail.setRiskFlags("C");
+
+        final OperationTemplateEntity operationTemplate = new OperationTemplateEntity();
+
+        final Operation result = tested.convert(operationDetail, operationTemplate);
+
+        assertNotNull(result.getUi());
+
+        final UiExtensions ui = result.getUi();
+        assertNull(ui.getFlipButtons());
+        assertEquals(true, ui.getBlockApprovalOnCall());
+        assertNull(ui.getPreApprovalScreen());
+    }
+
+    @Test
+    void testConvertUi_riskFlags_XCF() throws Exception {
+        final OperationDetailResponse operationDetail = new OperationDetailResponse();
+        operationDetail.setSignatureType(List.of(SignatureType.KNOWLEDGE));
+        operationDetail.setStatus(OperationStatus.APPROVED);
+        operationDetail.setRiskFlags("XCF");
+
+        final OperationTemplateEntity operationTemplate = new OperationTemplateEntity();
+
+        final Operation result = tested.convert(operationDetail, operationTemplate);
+
+        assertNotNull(result.getUi());
+
+        final UiExtensions ui = result.getUi();
+        assertEquals(true, ui.getFlipButtons());
+        assertEquals(true, ui.getBlockApprovalOnCall());
+        assertNotNull(ui.getPreApprovalScreen());
+
+        final PreApprovalScreen preApprovalScreen = ui.getPreApprovalScreen();
+        assertEquals(PreApprovalScreen.ScreenType.WARNING, preApprovalScreen.getType());
```

Comment:
@kober32 Just highlight, that in this case, you will get 
```json
{
  "flipButtons": true,
  "blockApprovalOnCall": true,
  "preApprovalScreen": {
    "type": "WARNING"
  }
}
```

Reply:
OK, just note that with this, the current app will fail to display the operation as other things in `preApprovalScreen` are required, so this needs changes in the app - I'm creating an issue